sos/policies/distros/redhat.py

```diff
@@ -430,11 +433,26 @@ def upload_sftp(self):
                                                        password=_token)
         raise Exception("Could not retrieve valid or anonymous credentials")
 
+    def check_file_too_big(self, archive):
+        size = os.path.getsize(archive)
+        # Lets check if the size is bigger than the limit.
+        # There's really no need to transform the size to Gb,
+        # so we don't need to call any size converter implemented
+        # in tools.py
+        if len(str(size)) >= 10 and (size >= self._max_size_request):
+            self.ui_log.error(
+                _("Size of archive is bigger than Red Hat Customer Portal "
+                  f"limit for uploads of {self._max_size_human_readable} via "
+                  "sos http upload. \n")
+                  )
+            self.upload_url = RH_SFTP_HOST
```

Comment:
If we're changing the upload target on users, we should tell them as much. The current message makes it sound like we're not going to do any upload because of the size limit.

Alternatively, this should raise an exception to be caught by the try block this is called from.

Reply:
The output with this change is:

```
Please send this file to your support representative.

Size of archive is bigger than Red Hat Customer Portal limit for uploads of 1Gb via sos http upload. 

Attempting upload to Red Hat Secure FTP
```

The difference is that before, users will get an error after attempting to upload the whole file, and then a switch to SFTP with the "Attempting upload to Red Hat Secure FTP" message.

Do I still need to add yet another "we are failing over to SFTP" to the code above?

Reply:
I think this logging is sufficient.

Reply:
That's fair. I had the order of logging for the upload target reversed in my head.